### PR TITLE
update ipfs libraries; and fix termination commands response

### PR DESCRIPTION
**Pull request by sudeepdino008**
* update ipfs libraries (including kubo to v0.20.0)
* accept ctrl-c or kill as termination commands for the server

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -20,16 +20,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
-      - name: Set up Go 1.18
+      - name: Set up Go 1.19
         uses: actions/setup-go@v3
         with:
-          go-version: 1.18.10
+          go-version: 1.19
```
Comment (noslav) [inline .github/workflows/golangci-lint.yml:26]: Are we now using `1.19` across the full network stack? It's very nice to have that consistency for validators.

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -20,16 +20,16 @@ jobs:
...
         id: go
       - run: go version
 
       - name: Lint
         run: |
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.51.2
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.53.3
```
Comment (noslav) [inline .github/workflows/golangci-lint.yml:32]: Could you also please do these for `bsp-agent` and `bsp-geth` ?

File: README.md [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
+    - [Upload a file](#upload-a-file)
+    - [Download content (given cid)](#download-content-given-cid)
+    - [Find the cid given some content](#find-the-cid-given-some-content)
+  - [Running ipfs-pinner server with docker](#running-ipfs-pinner-server-with-docker)
+    - [Docker Volume setup](#docker-volume-setup)
+    - [Port mapping setup](#port-mapping-setup)
+  - [Development](#development)
+    - [Generate go http client go bindings via openapi](#generate-go-http-client-go-bindings-via-openapi)
+  - [Known Issues](#known-issues)
+    - [permission issue](#permission-issue)
```
Comment (noslav) [inline README.md:17]: Can the readme titles be consistently capitalized? and line `19` for example.

File: README.md [hunk 4]
```diff
@@ -126,7 +128,7 @@ Now, we can run the container:
 ```bash
 docker container run --detach --name ipfs-pinner-instance \
        --volume /tmp/data/.ipfs/:/root/.ipfs/  \
-       -p 4001:4001 -p 3001:3001  \
```
Comment (noslav) [inline README.md:129]: bitswap removal?

File: binary/main.go [hunk 5]
```diff
@@ -86,7 +86,7 @@ func download(ctx context.Context, node pinner.PinnerNode) {
 	//ccid, err := cid.Parse("bafybeifzst7cbujrqemiulznrkttouzshnqkrajiib5fp5te53ojs5sl5u") // file encapsulated in folder
 	//ccid, err := cid.Parse("QmeFd8e4UaAPrPnwxBWcpqY3tMpggpWB3XYqftMpyYYLWZ") // straight up file
 
-	//ccid, err := cid.Parse("bafybeifzst7cbujrqemiulznrkttouzshnqkrajiib5fp5te53ojs5sl5u")
+	//ccid, err := cid.Parse("bfybeifzst7cbujrqemiulznrkttouzshnqkrajiib5fp5te53ojs5sl5u")
```
Comment (noslav) [inline binary/main.go:89]: there's an `a` missing after `b` from the start is that intended?

Comment (noslav) [reply binary/main.go:89]: can see that's not the case below?

File: car/gocar.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// A car exporter based on go-car module
+// A car exporter based on boxo/ipld/car module
```
Comment (noslav) [inline car/gocar.go:1]: why did we adopt `boxo`?

File: server/main.go [hunk 7]
```diff
@@ -96,12 +96,35 @@ func setUpAndRunServer(portNumber int, token, ipfsGatewayUrls string) {
...
 
 	log.Print("Listening...")
+	serverStopCommandSetup()
 	err := http.ListenAndServe(":"+strconv.Itoa(portNumber), mux)
 	if err != nil {
 		log.Println("error listening and serving on TCP network: %w", err)
 	}
 }
 
+func serverStopCommandSetup() {
```
Comment (noslav) [inline server/main.go:106]: Can you send the sigterm to the docker container and this be executed or do you have to exec in?

**Review by noslav — CHANGES_REQUESTED** [review]
Please address some changes necessary and some questions. Thanks!

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (sudeepdino008) [reply .github/workflows/golangci-lint.yml:26]: we are using 1.19 across the stack.

File: README.md (hunk 4 above)
Comment (sudeepdino008) [reply README.md:129]: yes. It's not used in any form right now. So just removing it to avoid confusion.

File: car/gocar.go (hunk 6 above)
Comment (sudeepdino008) [reply car/gocar.go:1]: ipfs moved a bunch of its code in a separate repo called boxo, it was just some refactoring from their end. The kubo repo now depends on boxo too, just like we need to now.

File: server/main.go (hunk 7 above)
Comment (sudeepdino008) [reply server/main.go:106]: let me check this

Comment (sudeepdino008) [reply server/main.go:106]: nope. Docker seems to use a different mechanism to kill the ipfs-pinner process.  
Anyway, this was meant for when running natively, as the process didn't get killed with ^C.

**Pull request merged**